src/components/legend/defaults.js

```diff
@@ -136,7 +180,7 @@ function groupDefaults(legendId, layoutIn, layoutOut, fullData) {
     coerce('groupclick');
 
     coerce('x', defaultX);
-    coerce('xanchor');
+    coerce('xanchor', defaultXAnchor);
     coerce('y', defaultY);
```

Comment:
We should be able to remove `coerce('x', xDefault)` and `coerce('y', yDefault)` now that we have the `Lib.coerce` calls above, right?